### ISSUE TITLE
TBB include headers and dyninst installation

### ISSUE THREAD

**Issue by pkestene** [dyninst/dyninst#730]
Hello,

[this issue is not about building TBB but using an already existing TBB]

I think there is probably a potential "hole" regarding the way TBB is treated in some corner cases :

I was building dyninst using an already existing installed version of TBB (version 2019.5), installed in a custom user directory.
So to build dyninst, I used cmake flag TBB_ROOT_DIR=/my/custom/path/to/tbb

Thus far, dyninst builds fine, and install fine, no problem.

But then I try to build a client code (e.g. Caliper), Caliper build system lost the information that dyninst was build using a speicific TBB version, for which the header location is not known (there is no include flags dedicated to tell where TBB headers are in Caliper, Caliper has no way to guess which TBB was used to build dyninst); even worse, if the system has another TBB installed (e.g. my Ubuntu 18.04 has an old version of TBB packaged in /usr); Caliper build fine because it will automatically use headers from /usr/include/tbb, that are the wrong TBB headers, i.e. not the ones use to build dyninst...

To summarize, I can live with it, I can manually change some flags to make sure Caliper sees the right TBB headers...

but I think that since dyninst uses cmake, all dyninst cmake targets that use TBB as a dependency, cmake should also explicitely declare
```cmake
target_include_directories(some_dyninst_target_name PUBLIC ${TBB_INCLUDE_DIR})
```  
so that TBB_INCLUDE_DIR can be transported to the client cmake build that uses it; at least that the installed DyninstConfig.cmake and others  (symtabAPITargets.cmake, etc...) should contain the information about TBB header location.

Correct me if I am wrong with this.

**Comment by hainest** [comment]
Dyninst's CMake build system is in an unfortunate state of being mostly pre-CMake 3.0 with a few sprinkles of "modern" (post-3.0) CMake. Modernizing the build system is on our list of things to do, but it's not likely to be completed soon. In that case, your solution would be the correct thing to do. The best workaround fortoday is to include the Dyninst CMake cache and extract the TBB_INCLUDE_DIR. This is how we handle this in our test suite (see https://github.com/dyninst/testsuite/blob/master/CMakeLists.txt#L23).

**Comment by hainest** [comment]
Closing due to inactivity. Please reopen or submit a new issue, if this problem persists.
